docs/configuredgraphfactory.txt

```diff
@@ -0,0 +1,457 @@
+
+[source, properties]
+----
+graphManager: org.janusgraph.graphdb.management.JanusGraphManager
+graphs {
+  graph1: conf/graph1.properties,
+  graph2: conf/graph2.properties
+}
+----
+
+but `conf/graph1.properties` and `conf/graph2.properties` do not include the property `graph.graphname`, then these graphs will be stored in the JanusGraphManager and thus bound in your gremlin script executions as `graph1` and `graph2`, respectively.
+
+
+[[important]]
+==== Important
+
+For convenience, if your configuration used to open a graph specifies `graph.graphname`, but does not specify the backend's storage directory, tablename, or keyspacename, then the relevant parameter will automatically be set to the value of `graph.graphname`. However, if you supply one of those parameters, that value will always take precedence. And if you supply neither, they default to the configuration option's default value.
+
+One special case is `storage.root` configuration option. This is a new configuration option used to specify the base of the directory that will be used for any backend requiring local storage directory access. If you supply this parameter, you must also supply the `graph.graphname` property, and the absolute storage directory will be equal to the value of the `graph.graphname` property appended to the value of the `storage.root` property.
+
+Below are some example use cases:
+
+1) Create a template configuration for my Cassandra backend such that each graph created using this configuration gets a unique keyspace equivalent to the `String` <graphName> provided to the factory:
+
+[source, gremlin]
+----
+Map map = new HashMap();
+map.put("storage.backend", "cassandrathrift");
+map.put("storage.hostname", "127.0.0.1");
+ConfiguredGraphFactory.createTemplateConfiguration(new
+MapConfiguration(map));
+
+def g1 = ConfiguredGraphFactory.create("graph1"); //keyspace === graph1
+def g2 = ConfiguredGraphFactory.create("graph2"); //keyspace === graph2
+def g3 = ConfiguredGraphFactory.create("graph3"); //keyspace === graph3
+----
+
+2) Create a template configuration for my BerkeleyJE backend such that each graph created using this configuration gets a unique storage directory equivalent to the "<storage.root>/<graph.graphname>":
+
+[source, gremlin]
+----
+Map map = new HashMap();
+map.put("storage.backend", "berkeleyje");
+map.put("storage.root", "/data/graphs");
+ConfiguredGraphFactory.createTemplateConfiguration(new
+MapConfiguration(map));
+
+def g1 = ConfiguredGraphFactory.create("graph1"); //storage directory === /data/graphs/graph1
+def g2 = ConfiguredGraphFactory.create("graph2"); //storage directory === /data/graphs/graph2
+def g3 = ConfiguredGraphFactory.create("graph3"); //storage directory === /data/graphs/graph3
+----
+
+[[examples]]
+=== Examples
+
+[source, gremlin]
+----
+gremlin> :remote connect tinkerpop.server conf/remote.yaml
+==>Configured localhost/127.0.0.1:8182
+
+gremlin> :> ConfiguredGraphFactory.open("graph");
```

Comment:
Can we drop the `:>` from the prompt and just keep the `gremlin> ` prefix?

Reply:
`:>` is required when using the Gremlin Console with a remote server connection. The alternative is to start a `:remote console` session after creating the remote connection.

Reply:
Changing to start a `:remote console` and dropping prefixes.